fab/fabfile.py

```diff
@@ -638,18 +645,95 @@ def _deploy_without_asking():
         # hard update of manifest.json since we're about to force restart
         # all services
         _execute_with_timing(update_manifest)
+        _execute_with_timing(clean_releases)
+    except PreindexNotFinished:
+        mail_admins(
+            " You can't deploy yet",
+            ("Preindexing is taking a while, so hold tight "
+             "and wait for an email saying it's done. "
+             "Thank you for using AWESOME DEPLOY.")
+        )
     except Exception:
         _execute_with_timing(mail_admins, "Deploy failed", "You had better check the logs.")
         # hopefully bring the server back to life
         _execute_with_timing(services_restart)
         raise
     else:
+        _execute_with_timing(update_current)
         _execute_with_timing(services_restart)
+        _execute_with_timing(record_successful_release)
         url = _tag_commit()
         _execute_with_timing(record_successful_deploy, url)
 
 
 @task
+@roles(ROLES_ALL_SRC)
+def update_current(release=None):
+    sudo('ln -nfs {} {}'.format(release or env.code_root, env.code_current))
+
+
+@task
+@roles(ROLES_ALL_SRC)
+def unlink_current():
+    if files.exists(env.code_current):
+        sudo('unlink {}'.format(env.code_current))
+
+
+@task
+@roles(ROLES_ALL_SRC)
+@parallel
+def create_code_dir():
+    sudo('mkdir -p {}'.format(env.code_root))
+
+
+@roles(ROLES_ALL_SRC)
+def copy_localsettings():
+    sudo('cp {}/localsettings.py {}/localsettings.py'.format(env.code_current, env.code_root))
```

Comment:
Should we not just link the localsettings? Just thinking about the workflow for updating localsettings via ansible.